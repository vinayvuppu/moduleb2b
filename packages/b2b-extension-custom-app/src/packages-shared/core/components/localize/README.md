# Localize

The `<Localize />` helps working with a
[`LocalizedString`](https://dev.commercetools.com/http-api-types.html#localizedstring).

It respects the user's `locale` and first attempts to find the language
specified there. If not present it walks through the list of available languages
and picks the first available one (first preference is English) until falling
back to the `fallback` provided.

## Examples

```js
<Localize object={props.product} objectKey="name" />
```

```js
<Localize object={props.product} objectKey="name" fallback="No localization" />
```

## Props

| Props       | Type     | Required | Values | Default | Description                                                                |
| ----------- | -------- | :------: | ------ | ------- | -------------------------------------------------------------------------- |
| `object`    | `object` |    ✅    | -      | -       | The object which on the `objectKey` might contain a `LocalizedString`      |
| `objectKey` | `string` |    ✅    | -      | -       | The key on the `object`                                                    |
| `fallback`  | `string` |    -     | -      | -       | The fallback used whenever the `object` does not have any localized string |
