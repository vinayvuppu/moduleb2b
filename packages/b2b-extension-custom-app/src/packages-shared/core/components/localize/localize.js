import PropTypes from 'prop-types';
import React from 'react';
import { withApplicationContext } from '@commercetools-frontend/application-shell-connectors';
import localize from '@commercetools-local/utils/localize';

export const Localize = props => (
  <span>
    {localize({
      obj: props.object,
      key: props.objectKey,
      language: props.language,
      fallbackOrder: props.languages,
      fallback: props.fallback,
    })}
  </span>
);
Localize.displayName = 'Localize';
Localize.propTypes = {
  object: PropTypes.object.isRequired,
  objectKey: PropTypes.string.isRequired,
  fallback: PropTypes.string,

  // HoC
  language: PropTypes.string.isRequired,
  languages: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default withApplicationContext(applicationContext => ({
  language: applicationContext.dataLocale,
  languages: applicationContext.project.languages,
}))(Localize);
