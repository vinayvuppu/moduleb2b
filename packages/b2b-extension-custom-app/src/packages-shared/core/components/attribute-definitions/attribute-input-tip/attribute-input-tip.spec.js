import React from 'react';
import { shallow } from 'enzyme';
import localize from '@commercetools-local/utils/localize';
import AttributeInputTip from './attribute-input-tip';

jest.mock('@commercetools-local/utils/localize', () =>
  jest.fn(() => 'localized value')
);

const render = props => shallow(<AttributeInputTip {...props} />);

const createTestProps = custom => ({
  language: 'de',
  languages: ['de', 'en'],
  ...custom,
});

describe('AttributeInputTip', () => {
  let wrapper;

  describe('attribute has input tip', () => {
    let props;

    beforeEach(() => {
      props = createTestProps({
        inputTip: {
          en: 'en input tip',
          de: 'de input tip',
        },
      });
      wrapper = render(props);
    });

    it('should localize tip', () => {
      expect(localize).toHaveBeenCalledWith({
        obj: { inputTip: props.inputTip },
        key: 'inputTip',
        language: props.language,
        fallback: '',
        fallbackOrder: props.languages,
      });
    });

    it('should render tip', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('attribute has no input tip', () => {
    beforeEach(() => {
      const props = createTestProps();
      wrapper = render(props);
    });

    it('should render nothing', () => {
      expect(wrapper).toBeEmptyRender();
    });
  });
});
