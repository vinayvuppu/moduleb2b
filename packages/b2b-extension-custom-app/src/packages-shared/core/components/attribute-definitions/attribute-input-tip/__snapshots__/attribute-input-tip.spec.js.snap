// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttributeInputTip attribute has input tip should render tip 1`] = `
<TextDetail>
  localized value
</TextDetail>
`;
